site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/blr-related-works/components/RecordItem.js

```diff
@@ -0,0 +1,126 @@
+// and/or modify it under the terms of the GNU General Public License as
+// published by the Free Software Foundation; either version 2 of the
+// License, or (at your option) any later version.
+//
+// Zenodo is distributed in the hope that it will be
+// useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Zenodo; if not, write to the
+// Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+// MA 02111-1307, USA.
+//
+// In applying this license, CERN does not
+// waive the privileges and immunities granted to it by virtue of its status
+// as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+import React from "react";
+import PropTypes from "prop-types";
+import { Grid, Header, Item, Label, Icon } from "semantic-ui-react";
+import _get from "lodash/get";
+import { SearchItemCreators } from "@js/invenio_app_rdm/utils";
+import { Image } from "react-invenio-forms";
+
+const layoutProps = (result) => ({
+  accessStatusId: _get(result, "ui.access_status.id", "open"),
+  accessStatus: _get(result, "ui.access_status.title_l10n", "Open"),
+  accessStatusIcon: _get(result, "ui.access_status.icon", "unlock"),
+  createdDate: _get(result, "ui.created_date_l10n_long", "Unknown date"),
+  creators: _get(result, "ui.creators.creators", []).slice(0, 3),
+  resourceType: _get(result, "ui.resource_type.title_l10n", "No resource type"),
+  title: _get(result, "metadata.title", "No title"),
+  link: _get(result, "links.self_html", "#"),
+  typeIcon: result.ui?.resource_type?.id.includes("image") ? "image" : "file alternate",
+});
+
+export const RecordGridItem = ({ result }) => {
+  const { title, creators, resourceType, typeIcon, link } = layoutProps(result);
+
+  return (
+    <Grid.Column className="flex column">
+      <Image
+        src={`${result.links.self_html}/thumb250`}
+        className="thumbnail-image mb-10"
+      />
+      <div>
+        <div className="pb-10">
+          <Label horizontal size="small">
+            <Icon name={typeIcon} />
+            {resourceType}
+          </Label>
+        </div>
+
+        <Header href={link} className="ui small header truncate-lines-2 m-0">
+          {title}
+        </Header>
+
+        <SearchItemCreators creators={creators} />
+      </div>
+    </Grid.Column>
+  );
+};
+
+RecordGridItem.propTypes = {
+  result: PropTypes.object.isRequired,
+};
+
+export const RecordListItem = ({ result }) => {
+  const {
+    accessStatusId,
+    accessStatus,
+    accessStatusIcon,
+    createdDate,
+    creators,
+    resourceType,
+    title,
+    typeIcon,
+    link,
+  } = layoutProps(result);
+
+  return (
+    <Item>
+      <Image
+        size="tiny"
+        src={`${result.links.self_html}/thumb250`}
```

Comment:
shelve: we should actually have this link generated in the service so that we don't have to manually construct it here.

Reply:
Referenced in https://github.com/zenodo/zenodo-rdm/issues/682